### PR TITLE
Remove trader

### PR DESCRIPTION
**Pull request by maciejzelaszczyk**
This removes the Trader component. An attempt at a call setting the payout account is made. What remains to be done is to: polish this call, add the possibility of passing this account via an env variable and add a script for actually paying the rewards out (from the point of view of the relayer).

### CODE REVIEW

File: relayer/relayer/src/main.rs [hunk 1]
```diff
@@ -274,6 +269,25 @@ async fn run_relayer(
...
+
+    let current_committee_id = most_azero
+        .current_committee_id(azero_connection.as_connection())
+        .await?;
+    most_azero
+        .set_payout_account(
+            &azero_signed_connection,
+            current_committee_id,
+            AccountId::from_string(&config.payout_address)
+                .map_err(|why| AzeroContractError::NotAccountId(why.to_string()))?,
```
Comment (Marcin-Radecki) [inline relayer/relayer/src/main.rs:287]: This should be backward compatible - let's return `None` if there's no  such entry `config`

File: relayer/scripts/entrypoint.sh [hunk 2]
```diff
@@ -143,33 +139,18 @@ if [[ -n "${ETH_MIN_CONFIRMATIONS}" ]]; then
...
-
 if [[ "${ETH_GAS_LIMIT}" =~ ^[0-9]+$ ]]; then
   echo "Setting --eth-gas-limit to ${ETH_GAS_LIMIT}"
   ARGS+=(--eth-gas-limit=${ETH_GAS_LIMIT})
 fi
 
+if [[ -n "${PAYOUT_ADDRESS}" ]]; then
+  ARGS+=(--payout-address=${PAYOUT_ADDRESS})
+else
+  echo "Payout address required but not provided." 1>&2
```
Comment (Marcin-Radecki) [inline relayer/scripts/entrypoint.sh:150]: Please make this change backward compatible

File: relayer/relayer/src/main.rs [hunk 3]
```diff
@@ -274,6 +269,25 @@ async fn run_relayer(
...
+        )
+        .await?;
```
Comment (kostekIV) [inline relayer/relayer/src/main.rs:289]: Pls extract to function

File: relayer/relayer/src/main.rs (hunk 1 above)
Comment (maciejzelaszczyk) [reply relayer/relayer/src/main.rs]: I have changed this such that the parameter is now optional.

File: relayer/relayer/src/main.rs (hunk 3 above)
Comment (maciejzelaszczyk) [reply relayer/relayer/src/main.rs]: Extracted.

**Review by kostekIV — APPROVED** [review]
lgtm

File: relayer/scripts/run_relayer.sh [hunk 4]
```diff
@@ -42,6 +42,13 @@ if [[ -n "${SIGNER_CID}" ]]; then
   ARGS+=(--signer-cid=${SIGNER_CID})
 fi
 
+if [[ -n "${PAYOUT_ADDRESS}" ]]; then
+  ARGS+=(--payout-address=${PAYOUT_ADDRESS})
+else
+  echo "Payout address required but not provided." 1>&2
+  exit 1
```
Comment (Marcin-Radecki) [inline relayer/scripts/run_relayer.sh:49]: Why do we treat lack of `PAYOUT_ADDRESS` as error condition here?

**Review by Marcin-Radecki — APPROVED** [review]

File: relayer/scripts/run_relayer.sh (hunk 4 above)
Comment (maciejzelaszczyk) [reply relayer/scripts/run_relayer.sh]: We shouldn't be doing this - I missed it. Fixed now.

**Pull request merged**